generator/internal/stub_generator.cc

```diff
@@ -221,58 +221,61 @@ Status StubGenerator::GenerateHeader() {
-    mixin_stub_inputs_list.push_back(absl::StrFormat(
-      "std::unique_ptr<%s::StubInterface> %s", mixin_grpc_stub.second, mixin_grpc_stub.first));
-    mixin_stub_members_init_list.push_back(absl::StrFormat(
-      "%s_(std::move(%s))", mixin_grpc_stub.first, mixin_grpc_stub.first));
-    mixin_stub_members_list.push_back(absl::StrFormat(
-      "std::unique_ptr<%s::StubInterface> %s_;", mixin_grpc_stub.second, mixin_grpc_stub.first));
-  }
-  std::string const mixin_stub_inputs = absl::StrJoin(mixin_stub_inputs_list, ",\n");
-  std::string const mixin_stub_members_init = absl::StrJoin(mixin_stub_members_init_list, ",\n");
-  std::string const mixin_stub_members = absl::StrJoin(mixin_stub_members_list, "\n");
-
   if (HasLongrunningMethod()) {
-    HeaderPrint(  // clang-format off
-    absl::StrCat(R"""(
+    HeaderPrint(R"""(
   Default$stub_class_name$(
       std::unique_ptr<$grpc_stub_fqn$::StubInterface> grpc_stub,
-  )"""
-      , mixin_stub_inputs, mixin_stub_inputs_list.empty() ? "" : ",\n",
-R"""(    std::unique_ptr<google::longrunning::Operations::StubInterface> operations)
+    )""");
+    for (auto const& mixin_grpc_stub : mixin_grpc_stubs) {
+      HeaderPrint(absl::StrFormat(
+      R"""(      std::unique_ptr<%s::StubInterface> %s,
+)""", mixin_grpc_stub.second, mixin_grpc_stub.first));
+    }
+    HeaderPrint(R"""(      std::unique_ptr<google::longrunning::Operations::StubInterface> operations)
       : grpc_stub_(std::move(grpc_stub)),
-)"""
-      , mixin_stub_members_init, mixin_stub_members_init_list.empty() ? "" : ",\n",
-R"""(        operations_(std::move(operations)) {}
-)"""));
-    // clang-format on
+)""");
+    for (auto const& mixin_grpc_stub : mixin_grpc_stubs) {
+      HeaderPrint(absl::StrFormat(
+      R"""(        %s_(std::move(%s)),
+)""", mixin_grpc_stub.first, mixin_grpc_stub.first));
+    }
+    HeaderPrint(R"""(        operations_(std::move(operations)) {}
+)""");
+
   } else {
-    HeaderPrint(  // clang-format off
-    absl::StrCat(R"""(
+    HeaderPrint(R"""(
   explicit Default$stub_class_name$(
-      std::unique_ptr<$grpc_stub_fqn$::StubInterface> grpc_stub)"""
-      , mixin_stub_inputs_list.empty() ? "" : ",\n", mixin_stub_inputs,
-      R"""()
-      : grpc_stub_(std::move(grpc_stub)))"""
-      , mixin_stub_members_init_list.empty() ? "" : ",\n", mixin_stub_members_init,
-      R"""( {}
-)"""));
-    // clang-format on
+      std::unique_ptr<$grpc_stub_fqn$::StubInterface> grpc_stub)""");
+    for (auto const& mixin_grpc_stub : mixin_grpc_stubs) {
+      HeaderPrint(absl::StrFormat(R"""(,
+      std::unique_ptr<%s::StubInterface> %s
+)""", mixin_grpc_stub.second, mixin_grpc_stub.first));
+    }
+    HeaderPrint(R"""()
+      : grpc_stub_(std::move(grpc_stub)))""");
+    for (auto const& mixin_grpc_stub : mixin_grpc_stubs) {
+      HeaderPrint(absl::StrFormat(R"""(,
+        %s_(std::move(%s)))""", mixin_grpc_stub.first, mixin_grpc_stub.first));
+    }
+    HeaderPrint(R"""( {}
+)""");
```

Comment:
This stuff should get printed in the source file, not in the header. The generated code is not going to compile.

Given that we are delaying turning on the mixins, I think we could just remove these code stanzas and add them back in the next PR which does the generation.

WDYT?

Reply:
https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/datamigration/v1/internal/data_migration_stub.h#L449-L455

The constructor originally in the header, and the generated libraries compiles when I tested. Is my understanding of your question correct?

Reply:
Oh, or if you mean there will be any mixin relevant changes appearing here before mixin is enabled? Then `mixin_grpc_stubs` will be empty, nothing will be added in the generated libraries. This can be verified by `generate-libraries-pr` check.

Reply:
> The constructor originally in the header

D'oh, sorry. I just assumed it was in the source.